src/os/bluestore/Allocator.h

```diff
@@ -56,11 +56,14 @@ class Allocator {
   virtual void dump() = 0;
   virtual void dump(std::function<void(uint64_t offset, uint64_t length)> notify) = 0;
 
-  virtual void zoned_set_zone_states(std::vector<zone_state_t> &&_zone_states) {}
-  virtual bool zoned_get_zones_to_clean(std::deque<uint64_t> *zones_to_clean) {
-    return false;
+  virtual void zoned_init_alloc(std::vector<zone_state_t> &&_zone_states,
```

Comment:
May be just apply dynamic_cast<ZonedAllocator> when we call these "zoned_" methods and do not pollute Allocator's interface with them? Looks like we have just a few auxiliary and zone-aware locations when we need that cast...

Reply:
Ugh. I don't like dynamic_cast, but if you insist :-). Do you think we still need zoned_ prefixes in Zoned{Allocator, FreelistManager} after this change?

Reply:
I can't say I love them too... But IMO this time it's worth it.
And yeah zoned_ prefixes seem redundant then. Up to you to make a final decision though.

Reply:
Removed prefixes.